pkg/availability/multisigcollector/internal/parts/certcreation/certcreation.go

```diff
@@ -101,32 +138,26 @@ func IncludeCreatingCertificates(
 			// Ignore invalid signature.
 			return nil
 		}
-		requestState, ok := state.RequestState[context.reqID]
+		certificate, ok := state.Certificate[context.reqID]
 		if !ok {
 			// The request has already been completed.
 			return nil
 		}
 
-		requestState.sigs[nodeID] = context.signature
-		return nil
-	})
+		certificate.sigs[nodeID] = context.signature
 
-	// When F+1 signatures are collected, create and output a certificate.
-	dsl.UponCondition(m, func() error {
-		// Iterate over active outgoing requests.
-		// Most of the time, there is expected to be at most one active outgoing request.
-		for reqID, requestState := range state.RequestState {
-			if len(requestState.sigs) >= params.F+1 {
-				certNodes, certSigs := maputil.GetKeysAndValues(requestState.sigs)
+		newDue := len(certificate.sigs) >= params.F+1 // keep this here...
```

Comment:
No, but the resulting bool is used after the line that changes the state. I cannot know if a new one is due after the call l to `sendIfReady` because if the certificates is ready to be sent then it deletes the certificate (and thus the check is not valid).

Reply:
I see... From the way the function is used, it would make sense to me to have the `sendIfReady` function return a `bool` saying whether a new cert should be created (or maybe even trigger it itself, but not sure about that). What do you think?

Reply:
`sendIfReady` should then check if that particular certificate is ready to be sent, and not if *any* is (as is the case now) in order to return the bool. 

If this was Rust, then keeping a reference (pointer) to the value should be enough, but in Go keeping a pointer to a value of a map is undefined. I think the most correct and easiest conceptually at the moment is to leave it as is.